trio/_path.py

```diff
@@ -65,20 +104,31 @@ async def wrapper(self, *args, **kwargs):
     return wrapper
 
 
-def classmethod_wrapper_factory(cls, meth_name):
-    @classmethod
+def classmethod_wrapper_factory(
+    cls: AsyncAutoWrapperType, meth_name: str
+) -> classmethod:
     @async_wraps(cls, cls._wraps, meth_name)
-    async def wrapper(cls, *args, **kwargs):
+    async def wrapper(cls: type[Path], *args: Any, **kwargs: Any) -> Path:
```

Comment:
I don't really like the `Any`s here, I presume paramspecs are hard to put in place? If so then eh, not a priority, no need to fix.

Reply:
In several of these functions the method isn't passed in at all, so there's nothing to bind the paramspec to. I didn't think it would be particularly useful to try defining a bunch of protocols/vars and rearranging this code to make it type safe. It's too dynamic for that to really validate anything, and these are private anyway.